src/circuit/gadgets/constraints.rs

```diff
@@ -427,6 +464,148 @@ pub fn must_be_simple_bit(x: &Boolean) -> AllocatedBit {
     }
 }
 
+// Enforce div and mod operation for U64. We need to show that
+// arg1 = q * arg2 + r, such that 0 <= r < arg2.
+pub fn enforce_u64_div_mod<F: LurkField, CS: ConstraintSystem<F>>(
+    mut cs: CS,
+    g: &GlobalAllocations<F>,
+    cond: Boolean,
+    arg1: &AllocatedPtr<F>,
+    arg2: &AllocatedPtr<F>,
```

Comment:
Is there any reason this operates on `AllocatedPtr` rather than `AllocatedNum`. That seems to distract from the fact that these values are being used a field elements. This in turn complicates testing, etc.

Reply:
done